src/store/reducers/popupReducer.ts

```diff
@@ -0,0 +1,56 @@
+import { RootState } from '../store';
+import { PopupActionType } from '../actions/popupActions';
+import { Item } from '../../types/item';
+import { Category } from '../../types/category';
+
+export type PopupPropsType = {
+  title: string;
+  item?: Item | Category;
+  closeHandler: () => void;
+  onSubmit: (...params: any[]) => void,
+};
+
+export enum PopupType {
+  CATEGORY_FORM = 'Category Form',
+  ITEM_FORM = 'Item Form',
+  LOGIN_CONFIRM = 'Login Confirm',
+  DELETE_CONFIRM = 'Delete Confirm',
+  NOTIFICATION_MESSAGE = 'Notification Message',
+}
+export interface PopupState {
+  popupKey: string;
+  popupProps: PopupPropsType;
+}
+
+type PopupAction = {
+  type: string;
+  payload?: PopupState;
+};
+
+export const initialState: PopupState = {
+  popupKey: '',
+  popupProps: {
+    title: '',
+    closeHandler: () => null,
+    onSubmit: () => null,
```

Comment:
```suggestion
    closeHandler: () => void,
    onSubmit: () => void,
```

Reply:
These line are implementation of function, not define function type. Cannot return void here

Reply:
So we should assign null, no need to define function here.